e2e/e2e_test.go

```diff
@@ -0,0 +1,220 @@
+
+	// set up coordinator and chains
+	t := s.T()
+	coord := types.NewCoordinator(t, 2)
+	provChain := coord.GetChain(ibctesting.GetChainID(1))
+	consChain := coord.GetChain(ibctesting.GetChainID(2))
+
+	s.Coordinator = coord
+	s.ConsumerChain = wasmibctesting.NewWasmTestChain(consChain)
+	s.ProviderChain = wasmibctesting.NewWasmTestChain(provChain)
+	s.ConsumerApp = consChain.App.(*app.ConsumerApp)
+	s.IbcPath = ibctesting.NewPath(consChain, provChain)
+	s.ProviderDenom = sdk.DefaultBondDenom
+	s.ConsumerDenom = sdk.DefaultBondDenom
+	s.MyProvChainActor = provChain.SenderAccount.GetAddress().String()
+
+	s.initialHeader = types.GenInitialBTCHeaderInfo()
+}
+
+func (s *BabylonSDKTestSuite) Test1ContractDeployment() {
+	// consumer client
+	consumerCli := types.NewConsumerClient(s.T(), s.ConsumerChain)
+	// setup contracts on consumer (now just fetches addresses)
+	consumerContracts, err := consumerCli.BootstrapContracts(s.initialHeader)
+	s.NoError(err)
+	// provider client
+	providerCli := types.NewProviderClient(s.T(), s.ProviderChain)
+
+	s.NotEmpty(consumerCli.Chain.ChainID)
+	s.NotEmpty(providerCli.Chain.ChainID)
+	s.NotEmpty(consumerContracts.Babylon)
+	s.NotEmpty(consumerContracts.BTCLightClient)
+	s.NotEmpty(consumerContracts.BTCStaking)
+	s.NotEmpty(consumerContracts.BTCFinality)
+
+	s.ProviderCli = providerCli
+	s.ConsumerCli = consumerCli
+	s.ConsumerContract = consumerContracts
+
+	// assert the contract addresses are updated in params
+	ctx := s.ConsumerChain.GetContext()
+	s.Equal(s.ConsumerContract.Babylon.String(), s.ConsumerApp.BabylonKeeper.GetBSNContracts(ctx).BabylonContract)
+	s.Equal(s.ConsumerContract.BTCLightClient.String(), s.ConsumerApp.BabylonKeeper.GetBSNContracts(ctx).BtcLightClientContract)
+	s.Equal(s.ConsumerContract.BTCStaking.String(), s.ConsumerApp.BabylonKeeper.GetBSNContracts(ctx).BtcStakingContract)
+	s.Equal(s.ConsumerContract.BTCFinality.String(), s.ConsumerApp.BabylonKeeper.GetBSNContracts(ctx).BtcFinalityContract)
+
+	// query admins
+	adminRespStaking, err := s.ConsumerCli.Query(s.ConsumerContract.BTCStaking, types.Query{"admin": {}})
+	s.NoError(err)
+	s.Equal(adminRespStaking["admin"], s.ConsumerCli.GetSender().String())
+	adminRespFinality, err := s.ConsumerCli.Query(s.ConsumerContract.BTCFinality, types.Query{"admin": {}})
+	s.NoError(err)
+	s.Equal(adminRespFinality["admin"], s.ConsumerCli.GetSender().String())
+}
+
+func (s *BabylonSDKTestSuite) Test2InsertBTCHeaders() {
+	// generate headers
+	headers, headersMsg := types.GenBTCHeadersMsg(s.initialHeader)
+	headersMsgBytes, err := json.Marshal(headersMsg)
+	s.NoError(err)
+	// send headers to the BTCLightClient contract. This is to ensure that the contract is
+	// indexing BTC headers correctly.
+	res, err := s.ConsumerCli.Exec(s.ConsumerContract.BTCLightClient, headersMsgBytes)
+	s.NoError(err, res)
+
+	// query the base header
+	baseHeader, err := s.ConsumerCli.Query(s.ConsumerContract.BTCLightClient, types.Query{"btc_base_header": {}})
```

Comment:
The naming `initial_header` is used on the contract side, I may rename it to `base_header` for consistency :(

Reply:
Sounds good, we can do this separately. In general so many things to rename :(